flake.nix

```diff
@@ -0,0 +1,59 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/25.05";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        ghcWithPkgs = pkgs.haskellPackages.ghcWithPackages (hpkgs: with hpkgs; [
+          colour
+          directory
+          evdev
+          extra
+          JuicyPixels
+          lucid2
+          optics
+          optparse-generic
+          pretty-simple
+          shake
+          streamly
+          text
+          unix
+        ]);
+      in
+      {
+        devShells.default = pkgs.mkShell {
+          buildInputs = with pkgs; [
+            ghcWithPkgs
+            fourmolu
+            haskell-language-server
+          ];
+        };
+      } // {
+        packages = pkgs.lib.listToAttrs (map
+          (name: {
+            inherit name;
+            value =
+              let module = pkgs.lib.strings.toUpper (pkgs.lib.substring 0 1 name) + pkgs.lib.substring 1 (-1) name;
+              in pkgs.stdenv.mkDerivation {
+                inherit name;
+                src = ./.;
+                buildInputs = [ ghcWithPkgs ];
+                buildPhase = ''
+                  ghc -O2 -o ${name} -main-is ${module} ${module}.hs
+                '';
+                installPhase = ''
+                  mkdir -p $out/bin
+                  cp ${name} $out/bin/
+                '';
+                meta.mainProgram = name;
+              };
+          }) [
+          "hello"
+          "magicMouse"
```

Comment:
The mouse script should probably just be moved to my NixOS repo, since that's now where it's used.

Reply:
In that context, we may also wish to make some adjustments, particularly to logging.